Orchard1/Reset-AppDataFolder/Reset-AppDataFolder.psm1

```diff
@@ -16,29 +16,33 @@ function Reset-AppDataFolder
     [Alias("rsad")]
     Param
     (
-        [Parameter(Mandatory=$true, ValueFromPipelineByPropertyName=$true, Position=0)]
-        [string] $Path = $(throw "You need to specify that path an App_Data folder."),
+        [Parameter(
+            Mandatory = $true,
+            ValueFromPipelineByPropertyName = $true,
+            Position = 0,
+            HelpMessage = "You need to specify that path an App_Data folder.")]
+        [string] $Path,
 
         [switch] $KeepLogFiles
     )
 
     Process
-    {        
-        if (!(Test-Path $Path))
+    {
+        if (-not (Test-Path $Path))
```

Comment:
Why is `-not` and `-isnot` better?

Reply:
They are the canonical comparison operators. `!` is not mentioned [in the official documentation](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_comparison_operators?view=powershell-7.2) as an operator at all. So to me `!` feels like an alias to `-not` (except it's worse becuase afaik it's undocumented).

The same applies to `-isnot`, and also it makes things simpler because you get the result with a single operator instead of the combination of a prefix and an infix operator. Since the operators are so bulky, I'm pretty happy that MS gave us `-isnot`, `-notlike`, `-notmatch`, etc.

Reply:
I see, thanks.